pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/service/actvityinstancebuilder/form/block/FormBlockCreatorHelper.java

```diff
@@ -74,20 +73,17 @@ ComponentBlock createComponentBlock(AIBuilderContext ctx, ComponentBlockDef comp
     }
 
     TabularBlock createTabularBlock(AIBuilderContext ctx, TabularBlockDef tabularBlockDef) {
-        final List<List<Question>> rows = new ArrayList<>();
-        for (int row = 0; row < tabularBlockDef.getRowsCount(); row++) {
-            final List<Question> questionsRow = new ArrayList<>();
-            for (int column = 0; column < tabularBlockDef.getColumnsCount(); column++) {
-                final QuestionDef questionDef = tabularBlockDef.get(row, column);
-                if (questionDef == null) {
-                    questionsRow.add(null);
-                    continue;
-                }
-
-                questionsRow.add(ctx.getAIBuilderFactory().getQuestionCreator().createQuestion(ctx, questionDef));
+        final List<QuestionBlock> allQuestions = new ArrayList<>();
+        for (QuestionBlockDef questionBlockDef : tabularBlockDef.getQuestionBlocks()) {
+            if (questionBlockDef == null) {
```

Comment:
Why would there be a null questionBlockDef in here?

Reply:
Better: Better: why are we creating question tabular block from question blocks? Couldn't we just add the columnSpan property to questionBlock or even to FormBlock?
I mean: at this time we are focusing on adding questions to tabular, but high-level, we could potentially at some point decide to add other kinds of blocks. As far as I can see, we just need that one additional property. (columnSpan). Will allow you to delete at least one file and some additional code. And will be more general!

Reply:
@MocanaAtBroad :
TabularBlock contains TabularHeader in addition to columnSpan and QuestionBlocks

Reply:
changed to capture and return as FormBlock's but we might need to update DB schema support later